### PR TITLE
[feat] Styling Status Filter

### PR DESCRIPTION
**Pull request by nehaahussain**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- created a component for the style of the opened status filter
- status filter is a single-select filter
- apply button changes color when the user selects a status to filter by
- added font details to `texts.ts`

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Screenshot 2024-11-15 at 7 23 22 PM

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- press status filter
- review the following folders/files:
StatusDropdown folder, styles folder, schema, Filter folder (index.tsx), FilterBar folder, 

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."
- use GSS for technology filter

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @itsliterallymonique

### CODE REVIEW

File: components/StatusDropdown/styles.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import styled from 'styled-components';
+import COLORS from '../../styles/colors';
+
+export const FilterDropdownStyles = styled.div`
+  position: relative;
+  background: ${COLORS.white};
+  width: 10.5rem;
+  border-radius: 0.5rem;
+  padding-bottom: 0.5rem;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:9]: delete padding-bottom

File: components/StatusDropdown/styles.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+`;
+
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:11]: Make a new styled div. This is so we can house all the content in one place and add the padding around it (so that if Josh ever changes it again it's easy to change the padding directly from the figma). Use the styled div to wrap around the content (check my comments in index.tsx)
```
export const FilterContentDiv = styled.div`
  display: flex;
  flex-direction: column;
  padding: 0.625rem 1.25rem 0.9rem 1.25rem;
  align-content: space-between;
`;
```

File: components/StatusDropdown/styles.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+export const OptionTitleStyles = styled.p<{ color: string }>`
+  background: ${({ color }) => `${color}10`};
+  border-radius: 0.3rem;
+  padding: 0.1rem 0.4rem;
+  display: inline-block;
+`;
+
+export const CheckboxStyles = styled.input`
+  cursor: pointer;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:20]: add `margin-left: 0rem;` for some reason the button is not directly aligned with the apply button which we want

File: components/StatusDropdown/styles.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  background: ${({ isActive }) =>
+    isActive
+      ? COLORS.electricBlue
+      : 'rgba(73, 116, 224, 0.50)'}; // Active vs Inactive color
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  cursor: ${({ isActive }) => (isActive ? 'pointer' : 'not-allowed')};
+  border: none;
+  margin: 1.4rem auto 0.5rem auto;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:37]: auto not 0.5rem (we don't need this margin since we have a margin for the div wrapped around all the content

Comment (itsliterallymonique) [reply components/StatusDropdown/styles.ts:37]: It should look like `margin: 1.4rem auto auto auto;`

File: components/StatusDropdown/styles.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  pointer-events: ${({ isActive }) => (isActive ? 'auto' : 'none')};
+`;
+
+export const ButtonStyles = styled.button`
+  color: rgba(46, 58, 89, 0.85);
+  background: ${COLORS.white};
+  border: none;
+  border-radius: 6.25rem;
+  display: flex;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:46]: move `display: flex;` to the top so it looks more standard

File: components/StatusDropdown/styles.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  align-items: center;
+  cursor: pointer;
+`;
+
+export const ExitStyles = styled.div`
+  margin-left: auto;
+  margin-right: 1.1rem;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:53]: remove this

File: components/StatusDropdown/styles.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  margin-bottom: auto;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:54]: remove this

File: components/StatusDropdown/styles.ts [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  background: ${COLORS.white};
+  border: none;
+  border-radius: 6.25rem;
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+`;
+
+export const ExitStyles = styled.div`
+  margin-left: auto;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:52]: Add:
`display: flex;
  align-self: center;`

File: components/StatusDropdown/styles.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  margin-right: 1.1rem;
+  margin-bottom: auto;
+`;
+
+export const ButtonWithIconStyles = styled.div`
+  display: flex;
+  align-items: center;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:59]: add `justify-content: space-between;`

File: components/StatusDropdown/styles.ts [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  gap: 0.5rem;
+  margin-left: 1rem;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:61]: delete

File: components/StatusDropdown/styles.ts [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+  padding-top: 1rem;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:62]: delete

File: components/StatusDropdown/styles.ts [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+  margin-bottom: -0.4rem;
+  cursor: pointer;
+`;
+
+export const CheckboxContainer = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 0.5rem;
+  margin-bottom: -1.5rem;
+  margin-left: 1rem;
```
Comment (itsliterallymonique) [inline components/StatusDropdown/styles.ts:72]: remove

File: components/StatusDropdown/index.tsx [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  ];
+
+  const handleStatusChange = (status: string) => {
+    setSelectedStatus(selectedStatus[0] === status ? [] : [status]);
+  };
+
+  const isApplyButtonActive = selectedStatus.length > 0;
+
+  return (
+    <FilterDropdownStyles>
```
Comment (itsliterallymonique) [inline components/StatusDropdown/index.tsx:50]: below this, add <FilterContentDiv> to wrap around all the filter content

File: components/StatusDropdown/index.tsx [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+                {' '}
+                {option.title}
+              </FilterOptionsText>
+            </OptionTitleStyles>
+          </CheckboxContainer>
+        ))}
+      </div>
+      <ApplyButtonStyles isActive={isApplyButtonActive}>
+        <ApplyFiltersText>APPLY</ApplyFiltersText>
+      </ApplyButtonStyles>
```
Comment (itsliterallymonique) [inline components/StatusDropdown/index.tsx:79]: below this, add </FilterContentDiv>

File: components/StatusDropdown/index.tsx (hunk 13 above)
Comment (itsliterallymonique) [reply components/StatusDropdown/index.tsx:50]: Should look like this:
Screenshot 2024-11-16 at 3 37 53 PM

**Review by itsliterallymonique — CHANGES_REQUESTED** [review]
Made comments on the changes you should make. Styling looks really good. I just suggested these changes to make it easier for us later on if the designer decides to adjust margins / makes changes to the design.

**Review by itsliterallymonique — APPROVED** [review]
great job you can squash and merge
